packages/language-server/src/hover.ts

```diff
@@ -0,0 +1,255 @@
+  };
+  return hoverItem;
+}
+
+function getUI5NodeByElement(
+  astNode: XMLElement,
+  model: UI5SemanticModel
+): UI5Class | UI5Aggregation | undefined {
+  const ui5Class = find(
+    model.classes,
+    (ui5class) => xmlToFQN(astNode) === ui5NodeToFQN(ui5class)
+  );
+
+  if (astNode.parent.type === "XMLDocument" || ui5Class != undefined) {
+    return getClassByElement(astNode, model);
+  } else {
+    const parentElementClass = getClassByElement(astNode.parent, model);
+    return findAggragationByName(parentElementClass, astNode.name);
+  }
+}
+
+function getUI5NodeByKey(
+  astNode: XMLAttribute,
+  model: UI5SemanticModel
+): UI5Prop | UI5Event | UI5Association | undefined {
+  const parentElementClass = getClassByElement(astNode.parent, model);
+  return findUI5ClassMemberByName(parentElementClass, astNode.key);
+}
+
+function getUI5NodeByValue(
+  astNode: XMLAttribute,
+  model: UI5SemanticModel
+): UI5Namespace | UI5EnumValue | undefined {
+  if (astNode.key != null && isXMLNamespaceKey(astNode.key)) {
+    const ui5Namespace = find(
+      model.namespaces,
+      (namespace) => astNode.value === ui5NodeToFQN(namespace)
+    );
+    return ui5Namespace;
+  } else {
+    const tag = astNode.parent;
+    const elementClass = getClassByElement(tag, model);
+    return getUI5EnumByElement(elementClass, astNode.key, astNode.value);
+  }
+
+  return undefined;
+}
+
+function findUI5ClassMemberByName(
+  ui5Class: UI5Class | undefined,
+  targetName: string | null
+): UI5Prop | UI5Event | UI5Association | undefined {
+  if (ui5Class != undefined) {
+    const allProps: (UI5Prop | UI5Event | UI5Association)[] = flattenProperties(
+      ui5Class
+    );
+    const allEvents = flattenEvents(ui5Class);
+    const allAssociations = flattenAssociations(ui5Class);
+    const allClassMembers = allProps.concat(allEvents).concat(allAssociations);
+    const found = find(allClassMembers, (_) => _.name === targetName);
+    return found;
+  }
+
+  return undefined;
+}
+
+function findAggragationByName(
+  ui5Class: UI5Class | undefined,
+  name: string | null
+): UI5Aggregation | undefined {
+  if (ui5Class != undefined) {
+    const allAggregations: UI5Aggregation[] = flattenAggregations(ui5Class);
+    const ui5Aggregation = find(
+      allAggregations,
+      (aggregation) => aggregation.name === name
+    );
+    return ui5Aggregation;
+  }
+
+  return undefined;
+}
+
+function getUI5EnumByElement(
+  ui5Class: UI5Class | undefined,
+  key: string | null,
+  value: string | null
+): UI5EnumValue | undefined {
+  if (ui5Class != undefined) {
+    const properties = flattenProperties(ui5Class);
+    const ui5Property = find(properties, ["name", key]);
+    const propType = ui5Property?.type;
+    if (propType?.kind !== "UI5Enum") {
+      return undefined;
+    }
+
+    const ui5Enum = find(propType.fields, ["name", value]);
+    return ui5Enum;
+  }
+
+  return undefined;
+}
+
+export function isXMLNamespaceKey(key: string): boolean {
```

Comment:
@tal-sapan is also creating utilities related to this in \@xml-tools, we should re-use them (when merged)

Reply:
See https://github.com/SAP/xml-tools/pull/173

Reply:
It is merged